### PR TITLE
feat: add documentation for preferences

### PR DESCRIPTION
**Pull request by ouuan**
The preferences can cover most features, so it's good to write documentation for it instead of "Components" and "Features".

### CODE REVIEW

File: content/en/docs/Preferences/General.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+title: "General"
```
Comment (coder3101) [inline content/en/docs/Preferences/General.md:2]: I think General should be in top of the list. Currently it is at very bottom. For very new users who are likely to follow URL from top-down they will read General in the end. Which is a pre-requisite for most of other sections. So I suggest to move it up in the list and if possible it should be renamed to General Information so people don't get confused if it's General Preference

File: content/en/docs/Preferences/Extensions.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+The path to the `clang-format` executable, or simply `clang-format` if it's in the `PATH` environment variable.
+
+### Style
+
+The Clang Format style. The content here is the same as the content of a `.clang-format` file.
+
+You can read [Clang-Format Style Options documentation](https://clang.llvm.org/docs/ClangFormatStyleOptions.html) or search for other materials to learn how to write the styles.
+
+## Language Server
```
Comment (coder3101) [inline content/en/docs/Preferences/Extensions.md:20]: You can provide a URL to LSP setup section from here. Maybe add a paragraph saying "For setting up Language Server to lint you code, please read this"

File: content/en/docs/Preferences/General.md (hunk 1 above)
Comment (ouuan) [reply content/en/docs/Preferences/General.md:2]: When I learn something I need right now, I don't learn all pre-requisites at first, I search for what I need, and click the links/search for the pre-requisites later.

That's like when you can't solve a CP problem, you don't learn the trick first, you read the tutorial first, and then find you don't know a trick, and then go to learn it.

Comment (ouuan) [reply content/en/docs/Preferences/General.md:2]: > Which is a pre-requisite for most of other sections.

I don't think so. Only a few settings need them.

Comment (coder3101) [reply content/en/docs/Preferences/General.md:2]: Ok

**Pull request merged**